### PR TITLE
MindsDB Inference Handler improvements

### PR DESCRIPTION
**Pull request by MinuraPunchihewa**
## Description

This PR improves the MindsDB Inference handler in order to make it Enterprise-ready.

## Type of change

- [X] 🐛 Bug fix (non-breaking change which fixes an issue)

## Verification Process

To ensure the changes are working as expected:

 - [X]   Test Location: `tests/unit/ml_handlers/test_mindsdb_inference.py` for unit tests and `mindsdb/integrations/handlers/mindsdb_inference/tests/test_mindsdb_inference_handler.py` for integration tests.
 - [X]   Verification Steps: Run the above test modules (for integration tests, a valid API key will need to be set to the `MDB_TEST_MDB_INFERENCE_API_KEY` env variable).

- [X] My code follows the style guidelines(PEP 8) of MindsDB.
- [X] I have appropriately commented on my code, especially in complex areas.
- [X] Necessary documentation updates are either made or tracked in issues.
- [X] Relevant unit and integration tests are updated or added.

### CODE REVIEW

File: tests/unit/ml_handlers/test_mindsdb_inference_integrations_tests.py
Comment (ZoranPandovski) [inline tests/unit/ml_handlers/test_mindsdb_inference_integrations_tests.py:7]: Should we keep the location for this type of test inside the unit?

**Review by ZoranPandovski — CHANGES_REQUESTED** [review]
Minor comment added. Also please check the https://github.com/mindsdb/mindsdb/actions/runs/8815465631/job/24197483314?pr=9099#step:6:119

File: tests/unit/ml_handlers/test_mindsdb_inference_integrations_tests.py
Comment (MinuraPunchihewa) [reply tests/unit/ml_handlers/test_mindsdb_inference_integrations_tests.py:7]: @ZoranPandovski I don't think so either, but if you take a look at all of the other tests in `unit/ml_handlers`, they are actually integration tests.
Should we maybe decide where we are going to place the integration and unit tests for ML handlers (temporarily until we  come to a final decision)?
I guess with the data handlers, we know this already; unit tests in `unit/handlers` and integration tests in the specific handler directories.

**Comment by ZoranPandovski** [comment]
@MinuraPunchihewa Please just check the lining bug before we merge this.

**Pull request merged**
